source/common/upstream/thread_aware_lb_impl.cc

```diff
@@ -141,6 +144,12 @@ ThreadAwareLoadBalancerBase::LoadBalancerImpl::chooseHost(LoadBalancerContext* c
     return nullptr;
   }
 
+  HostConstSharedPtr host;
+  if (host = LoadBalancerContextBase::selectOverrideHost(cross_priority_host_map_.get(), context);
```

Comment:
I would either initialize host outside the if-block, or move the declaration inside. Right now I'm not seeing much benefit to this pattern

Reply:
get it.